great_expectations/data_context/types/base.py

```diff
@@ -1319,6 +1320,66 @@ def config_version(self):
         return self._config_version
 
 
+class LegacyCheckpointConfig(BaseYamlConfig):
+    def __init__(
+        self,
+        module_name: Optional[str] = None,
+        class_name: Optional[str] = None,
+        validation_operator_name: Optional[str] = None,
+        batches: Optional[list] = None,
+        commented_map: Optional[CommentedMap] = None,
+    ):
+        super().__init__(commented_map=commented_map)
+        self._module_name = module_name or "great_expectations.checkpoint"
+        self._class_name = class_name or "LegacyCheckpointConfig"
+        self._validation_operator_name = (
+            validation_operator_name or "action_list_operator"
+        )
+        self._batches = batches or []
+
+    @property
+    def module_name(self):
+        return self._module_name
+
+    @property
+    def class_name(self):
+        return self._class_name
+
+    @property
+    def validation_operator_name(self):
+        return self._validation_operator_name
+
+    @property
+    def batches(self):
+        return self._batches
+
+    @classmethod
+    def get_config_class(cls):
+        raise cls  # LegacyCheckpointConfig
+
+    @classmethod
+    def get_schema_class(cls):
+        raise LegacyCheckpointConfigSchema
+
+
+class LegacyCheckpointConfigSchema(Schema):
+    class Meta:
+        unknown = INCLUDE
+
+    class_name = fields.Str()
+    module_name = fields.Str()
+    validation_operator_name = fields.Str()
+    batches = fields.List(
+        fields.Dict(
+            keys=fields.Str(validate=OneOf(["batch_kwargs", "expectation_suite_names"]))
+        )
+    )
+
+    @post_load
+    def make_legacy_checkpoint_config(self, data, **kwargs):
```

Comment:
@roblim I am wondering -- why not do it the "new checkpoint store" way.  Basically, all checkpoint_config.yml files would be managed by the checkpoint_store which writes into the base directory "checkpoints".  Hence, since we now have Config & Schema for the LegacyCheckpoint, we can manage the read/write for the LegacyCheckpointConfig the same way as for the new CheckpointConfig by using checkpoint_store.set(key, value) and checkpoint_store.get(key) -- just like in "tests/data_context/store/test_checkpoint_store.py".  What do you think?  Now that my PR is merged, you can update the implementation if you agree.  Or we can merge this and do it in another pass (either of us can do it).  Let us decide in the morning.  Thanks!

Reply:
Agreed